src/shibuya.js

```diff
@@ -21,60 +28,43 @@ const main = async () => {
-
-    console.log(`\na) Add a proxy of Turing (paraId:${TuringDev.paraId}) for Alice on Shibuya ...\n Proxy address: ${proxyOnShibuya}\n`);
-    await sendExtrinsic(shibuyaHelper.api, shibuyaHelper.api.tx.proxy.addProxy(proxyOnShibuya, 'Any', 0), keyPair);
-
-    console.log('\nb) Topping up the proxy account on Shibuya with SBY ...\n');
-    const sbyAmount = new BN(1000, 10);
-    const sbyAmountBN = sbyAmount.mul(getDecimalBN(shibuyaHelper.getDecimalBySymbol('SBY')));
-
-    const transferSBY = shibuyaHelper.api.tx.balances.transfer(proxyOnShibuya, sbyAmountBN.toString());
-    await sendExtrinsic(shibuyaHelper.api, transferSBY, keyPair);
+    const proxiesOnShibuya = await shibuyaHelper.getProxies(shibuyaAddress);
+    if (!_.find(proxiesOnShibuya, { delegate: keyring.encodeAddress(proxyOnShibuya, Shibuya.ss58), proxyType: 'DappsStaking' })) {
+        console.log(`\n Add a proxy of Turing (paraId:${TuringDev.paraId}) for Alice on Shibuya ...\n Proxy address: ${proxyOnShibuya}\n`);
+        await sendExtrinsic(shibuyaHelper.api, shibuyaHelper.api.tx.proxy.addProxy(proxyOnShibuya, 'DappsStaking', 0), keyPair);
+    }
+
+    const minBalance = new BN(MIN_BALANCE_IN_PROXY).mul(sbyDecimalBN);
+    const balance = await shibuyaHelper.getBalance(proxyOnShibuya);
+    if (balance.free.lt(minBalance)) {
+        console.log('\nb) Topping up the proxy account on Shibuya with SBY ...\n');
+        const sbyAmount = new BN(1000, 10);
+        const sbyAmountBN = sbyAmount.mul(sbyDecimalBN);
+        const transferSBY = shibuyaHelper.api.tx.balances.transfer(proxyOnShibuya, sbyAmountBN.toString());
+        await sendExtrinsic(shibuyaHelper.api, transferSBY, keyPair);
+    }
 
     console.log('\n2. One-time proxy setup on Turing');
+    console.log(`\na) Add a proxy for Alice on Turing If there is no proxy of Shibuya (paraId:${Shibuya.paraId})\n`);
     const proxyOnTuring = turingHelper.getProxyAccount(Shibuya.paraId, turingAddress);
-
-    console.log(`\na) Add a proxy of Shibuya (paraId:${Shibuya.paraId}) for Alice on Turing ...\nProxy address: ${proxyOnTuring}\n`);
-    await sendExtrinsic(turingHelper.api, turingHelper.api.tx.proxy.addProxy(proxyOnTuring, 'Any', 0), keyPair);
-
-    console.log('\nb) Topping up the proxy account on Turing with TUR ...\n');
-    const transferExtrinsic = turingHelper.api.tx.balances.transfer(proxyOnTuring, '10000000000000');
-    await sendExtrinsic(turingHelper.api, transferExtrinsic, keyPair);
+    const proxiesOnTuring = await turingHelper.getProxies(turingAddress);
+    if (!_.find(proxiesOnTuring, { delegate: keyring.encodeAddress(proxyOnTuring, TuringDev.ss58), proxyType: 'Any' })) {
+        console.log(`\n Add a proxy of Shibuya (paraId:${Shibuya.paraId}) for Alice on Turing ...\nProxy address: ${proxyOnTuring}\n`);
+        await sendExtrinsic(turingHelper.api, turingHelper.api.tx.proxy.addProxy(proxyOnTuring, 'Any', 0), keyPair);
```

Comment:
A reminder to update Any to the new proxy type.

Reply:
Yes, shibuya's proxy is DappsStaking.